PSPNet/train.py

```diff
@@ -123,33 +149,48 @@ def main():
 
                 if itr % 500 == 0 and itr > 0:
                     num_batches_per_epoch = (FLAGS.num_samples_per_val // FLAGS.batch_size) + 1
-                    val_losses, val_ious, val_px_accs = [], [], []
+                    val_losses = []
+
+                    # Reset mIoU and pixel accuracy running vars so we get a fresh evaluation
+                    # for each validation
+                    net.sess.run(running_metrics_init)
+
                     for i in range(num_batches_per_epoch):
 
                         val_images, val_annotations = net.sess.run([valid_image_batch, valid_label_batch])
-                        val_feed_dict = {image: val_images, annotation: val_annotations, keep_prob: 1.0, is_training: False}
+                        val_feed_dict = {image: val_images,
+                                         annotation: val_annotations,
+                                         keep_prob: 1.0, is_training: False}
 
-                        val_loss, val_predict = net.sess.run([loss, net.predict], feed_dict=val_feed_dict)
+                        val_loss, _ = net.sess.run([loss, metrics_update_op], feed_dict=val_feed_dict)
                         val_losses.append(val_loss)
 
-                        val_iou = compute_batch_iou(val_predict, val_annotations)
-                        val_ious.append(val_iou)
-
-                        val_px_acc = compute_batch_pixel_accuracy(val_predict, val_annotations)
-                        val_px_accs.append(val_px_acc)
+                    avg_val_loss = np.mean(val_losses)
+                    best_loss_val = net.sess.run(best_loss)
 
-                    avg_val_loss, avg_val_iou, avg_val_px_acc = np.mean(val_losses), np.mean(val_ious), np.mean(val_px_accs)
-                    logging.info("{} ---> Validation_loss: {} (best: {}), Validation_IoU: {} (best: {}), Validation pixel accuracy: {} (best: {})".format(
-                                    datetime.datetime.now(), avg_val_loss, best_loss.eval(session=net.sess), avg_val_iou, best_iou.eval(session=net.sess), avg_val_px_acc, best_px_acc.eval(session=net.sess)))
+                    # Get current and best mean IoUs and pixel accuracies
+                    val_iou, best_iou_val = net.sess.run([mIoU, best_iou])
+                    val_px_acc, best_px_acc_val = net.sess.run([px_acc, best_px_acc])
 
-                    best_loss_val = net.sess.run(best_loss)
+                    logging.info(
+                        "{} ---> Validation_loss: {} (best: {}), Validation_IoU: {} (best: {}), "
+                        "Validation pixel accuracy: {} (best: {})".format(
+                                    datetime.datetime.now(),
+                                    avg_val_loss, best_loss.eval(session=net.sess),
```

Comment:
why are you evaluating `best_loss` again, isn't it the same as best_loss_val on line 169?

Reply:
good call; moved eval up to be inline with running metrics eval, but forgot to change below; fixed.